matter_server/client/models/node.py

```diff
@@ -155,91 +173,64 @@ def set_attribute_value(self, attribute_path: str, attribute_value: Any) -> None
         )
         setattr(cluster_instance, attribute_name, attribute_value)
 
+    def update(self, attributes_data: dict[str, Any]) -> None:
+        """Update MatterEndpoint from (endpoint-specific) raw Attributes data."""
+        # unwrap cluster and clusterattributes from raw node data attributes
+        for attribute_path, attribute_value in attributes_data.items():
+            self.set_attribute_value(attribute_path, attribute_value)
+        # extract device types from Descriptor Cluster
+        if cluster := self.get_cluster(Clusters.Descriptor):
+            for dev_info in cluster.deviceTypeList:  # type: ignore[unreachable]
```

Comment:
Why would we do this every time?

Reply:
no only once

Reply:
that update is only called if the entire node updates (e.g. after interview)